pipeline/transformer/transformer_sierra/src/main/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/subjects/SierraSubjectsTransformer.scala

```diff
@@ -0,0 +1,53 @@
+package uk.ac.wellcome.platform.transformer.sierra.transformers.subjects
+
+import uk.ac.wellcome.platform.transformer.sierra.transformers.{
+  MarcUtils,
+  SierraTransformer
+}
+import uk.ac.wellcome.models.transformable.sierra.SierraBibNumber
+import uk.ac.wellcome.platform.transformer.sierra.source.{
+  SierraBibData,
+  VarField
+}
+import uk.ac.wellcome.models.work.internal.{
+  AbstractRootConcept,
+  MaybeDisplayable,
+  Subject
+}
+
+trait SierraSubjectsTransformer extends SierraTransformer with MarcUtils {
+
+  type Output = List[
+    MaybeDisplayable[
+      Subject[
+        MaybeDisplayable[AbstractRootConcept]
+      ]
+    ]
+  ]
+
+  val subjectVarFields: List[String]
+
+  def apply(bibId: SierraBibNumber, bibData: SierraBibData) =
+    getSubjectsFromVarFields(
+      bibId,
+      subjectVarFields.flatMap(getMatchingVarFields(bibData, _))
+    )
+
+  def getSubjectsFromVarFields(bibId: SierraBibNumber,
+                               varFields: List[VarField]): Output
+
+  /** Given a varField and a list of subfield tags, create a label by
+    * concatenating the contents of every subfield with one of the given tags.
+    *
+    * The order is the same as that in the original MARC.
+    *
+    */
+  def createLabel(varField: VarField, subfieldTags: List[String]): String =
+    varField.subfields
+      .filter { subfield =>
+        subfieldTags.contains(subfield.tag)
+      }
+      .sortBy { _.tag }
+      .map { _.content }
+      .mkString(" ")
```

Comment:
I thought we weren't meant to be adding any values? Does the marc fields not already contain ` `?

Reply:
Not sure I understand. This method already existed [here](https://github.com/wellcometrust/catalogue/blob/master/pipeline/transformer/transformer_sierra/src/main/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/subjects/SierraOrganisationSubjects.scala#L76-L77), I have just moved it up to the base trait.

It concatenates the content of various subfields into a single label (which we seem to do all over the place in the transformer)